src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2821,245 +2821,176 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
-            auto tx1Entries =
-                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
-            addTxn(-1, tx1Entries);
-            checkPrefetch(tx1Entries);
+        // Ensure that the correct entries were loaded.
+        for (auto const& k : expectedSuccessKeys)
+        {
+            REQUIRE(ltx2.load(k));
         }
+
+        auto numLoadedHot = root.prefetchSoroban(keysToPrefetch, lkMeterHot);
+        REQUIRE(numLoadedHot == 0);
+        // 100% hit rate but make it floating point
+        REQUIRE(fabs(ltx2.getPrefetchHitRate() - 1.0f) < 0.0001f);
     };
 
-    SECTION("prefetch with large cache")
+    std::vector<LedgerEntry> allEntries;
+    SECTION("all keys have quota")
     {
-        // cache size > number of entries
-        runTest(1000);
+        auto tx1Entries =
+            std::vector<LedgerEntry>{classicEntry, contractDataEntry};
+        auto tx2Entries =
+            std::vector<LedgerEntry>{classicEntry, contractDataEntry};
+        addTxn(true /* enough quota */, tx1Entries);
+        addTxn(false, tx2Entries);
+        allEntries.insert(allEntries.end(), tx1Entries.begin(),
+                          tx1Entries.end());
+        allEntries.insert(allEntries.end(), tx2Entries.begin(),
+                          tx2Entries.end());
+        std::set<LedgerKey> expectedSuccessKeys;
+        for (auto const& e : allEntries)
+        {
+            expectedSuccessKeys.emplace(LedgerEntryKey(e));
+        }
+        // No keys should fail.
+        std::set<LedgerKey> expectedFailedKeys;
+        checkPrefetch(allEntries, expectedSuccessKeys, expectedFailedKeys);
+    }
+
+    SECTION("dead keys don't affect quota")
+    {
+        auto deadKeys =
+            std::vector<LedgerKey>{deadKeySet.begin(), deadKeySet.end()};
+        auto tx1Entries =
+            std::vector<LedgerEntry>{classicEntry, contractDataEntry};
+        addTxn(true /* enough quota */, tx1Entries, deadKeys);
+        allEntries.insert(allEntries.end(), tx1Entries.begin(),
+                          tx1Entries.end());
+        std::set<LedgerKey> expectedSuccessKeys;
+        std::set<LedgerKey> expectedFailedKeys;
+        for (auto const& e : allEntries)
+        {
+            expectedSuccessKeys.emplace(LedgerEntryKey(e));
+        }
+        checkPrefetch(allEntries, expectedSuccessKeys, expectedFailedKeys);
     }
-    SECTION("prefetch with small cache")
+
+    SECTION("don't load entries without quota")
     {
-        // cache size < number of entries
-        runTest(1);
+        auto tx1Entries =
+            std::vector<LedgerEntry>{classicEntry, contractDataEntry, TTLEntry};
+        addTxn(false /* enough quota */, tx1Entries);
+        std::set<LedgerKey> expectedSuccessKeys;
+        std::set<LedgerKey> expectedFailedKeys;
+        expectedSuccessKeys.emplace(LedgerEntryKey(TTLEntry));
+        // Whichever entry is loaded first will succeed. The other will fail.
+        if (classicEntry < contractDataEntry)
```

Comment:
Technically classic entries are always strictly less than Soroban entry types, so you can just leave a comment that says the smallest entry will get loaded first successfully and point in the classicEntry instead of doing this if check.